RevenueCatUI/Scripts/PaywallResult.cs

```diff
@@ -0,0 +1,112 @@
+using System;
+
+namespace RevenueCat.UI
+{
+    /// <summary>
+    /// Represents the result of a paywall presentation.
+    /// </summary>
+    [Serializable]
+    public class PaywallResult
+    {
+        /// <summary>
+        /// The type of result from the paywall presentation.
+        /// </summary>
+        public PaywallResultType Result { get; }
+
+        /// <summary>
+        /// Creates a new PaywallResult.
+        /// </summary>
+        /// <param name="result">The result type</param>
+        public PaywallResult(PaywallResultType result)
+        {
+            Result = result;
+        }
+
+        // Convenient static properties for common results
+        [System.Obsolete("Use NotPresented instead")] 
+        public static PaywallResult NotNeeded => new PaywallResult(PaywallResultType.NotPresented);
+        public static PaywallResult NotPresented => new PaywallResult(PaywallResultType.NotPresented);
+        public static PaywallResult Cancelled => new PaywallResult(PaywallResultType.Cancelled);
+        public static PaywallResult Error => new PaywallResult(PaywallResultType.Error);
+        public static PaywallResult Purchased => new PaywallResult(PaywallResultType.Purchased);
+        public static PaywallResult Restored => new PaywallResult(PaywallResultType.Restored);
```

Comment:
do we actually need these static properties for anything? Maybe they can be internal